### DISCUSSION TITLE
Output TIFF image is blank after Image Magick upgrade

### DISCUSSION THREAD

**Discussion by huynhhuuhanh** [ImageMagick/ImageMagick#3078] [Help]
Hi team.
We're utilizing Image Magick in our project to convert images and have a plan to upgrade to version 7.0.10-35.
I have a tiff file, 
[file.zip](https://github.com/ImageMagick/ImageMagick/files/5773763/file.zip),
after convert to tiff, the output file is blank.
I've tried to figure out the cause, the input file has some rows that have no data at the end of the strip. 
While decoding all rows per strip, it catches premature EOF and ignores all decoded data before.
Can you please support to stop decoding data when it catches premature EOF, keep decoded data, and render them to the output file?
Thanks.

**Comment by fmw42** [comment]
Your Tiff file seems to be malformed. But the message is just a warning and so it still tries to process it. 


When I process it

`identify file.tiff`

file.tiff TIFF 1728x2268 1728x2268+0+0 1-bit Bilevel Gray 35485B 0.000u 0:00.002

**identify: incorrect count for field "PageNumber", expected 2, got 1**. 

`TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/984.

**Reply by huynhhuuhanh** [reply]
Thank you for your quick response.
I'd known that but most of the image viewers can open this file correctly.
The current version (7.0.8-14) we're using also can process this file well.
I've tried to stop decoding when reaching the end of data, just render decoded data, it's the same as the behavior of the current version.
Can you please consider this solution?
Thanks.

**Comment by fmw42** [comment]
Your Tiff file seems to be malformed. But the message is **just a warning** and **so it still processes your file**.

**Comment by snibgo** [comment]
IM 7.0.7-64  gives warnings and processes the file, and the result looks good.

IM 7.0.10-45 gives warnings and processes the file, but the result is entirely black.

I suppose 7.0.10-45 has a bug: it should either process the image correctly, or raise an error.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Reply by huynhhuuhanh** [reply]
Thank you very much.
